api_shield_operations.go

```diff
@@ -0,0 +1,221 @@
+//
+// See API documentation https://developers.cloudflare.com/api/operations/api-shield-endpoint-management-retrieve-information-about-an-operation
+type APIShieldGetOperationParams struct {
+	// Features represents a set of features to return in `features` object when
+	// performing making read requests against an Operation or listing operations.
+	Features []string
+}
+
+// APIShieldGetOperationsParams represents the query parameters to set when retrieving operations
+//
+// See API documentation https://developers.cloudflare.com/api/operations/api-shield-endpoint-management-retrieve-information-about-all-operations-on-a-zone
+type APIShieldGetOperationsParams struct {
+	// Features represents a set of features to return in `features` object when
+	// performing making read requests against an Operation or listing operations.
+	Features []string
+	// Direction to order results.
+	Direction string
+	// OrderBy when requesting a feature, the feature keys are available for ordering as well, e.g., thresholds.suggested_threshold.
+	OrderBy string
+	// Filters to only return operations that match filtering criteria, see APIShieldGetOperationsFilters
+	Filters *APIShieldGetOperationsFilters
+	// Pagination options to apply to the request.
+	Pagination *PaginationOptions
+}
+
+// APIShieldGetOperationsFilters represents the filtering query parameters to set when retrieving operations
+//
+// See API documentation https://developers.cloudflare.com/api/operations/api-shield-endpoint-management-retrieve-information-about-all-operations-on-a-zone
+type APIShieldGetOperationsFilters struct {
+	// Host filters results to only include the specified hosts.
+	Hosts []string
+	// Host filters results to only include the specified methods.
+	Methods []string
+	// Endpoint filter results to only include endpoints containing this pattern.
+	Endpoint string
+}
+
+// APIShieldGetOperationResponse represents the response from the api_gateway/operations/{id} endpoint.
+type APIShieldGetOperationResponse struct {
+	Result APIShieldOperation `json:"result"`
+	Response
+}
+
+// APIShieldGetOperationsResponse represents the response from the api_gateway/operations endpoint.
+type APIShieldGetOperationsResponse struct {
+	Result     []APIShieldOperation `json:"result"`
+	ResultInfo `json:"result_info"`
+	Response
+}
+
+// APIShieldDeleteOperationResponse represents the response from the api_gateway/operations/{id} endpoint (DELETE).
+type APIShieldDeleteOperationResponse struct {
+	Result interface{} `json:"result"`
+	Response
+}
+
+// GetAPIShieldOperation returns information about an operation
+//
+// API documentation https://developers.cloudflare.com/api/operations/api-shield-endpoint-management-retrieve-information-about-an-operation
+func (api *API) GetAPIShieldOperation(ctx context.Context, rc *ResourceContainer, operationID string, params *APIShieldGetOperationParams) (*APIShieldOperation, error) {
```

Comment:
the CRUD methods should only ever take a maximum of 3 parameters (see https://github.com/cloudflare/cloudflare-go/blob/master/docs/conventions.md#methods)

Reply:
fixed in https://github.com/cloudflare/cloudflare-go/pull/1397/commits/583a0d227aeb5d06bbfbb9d9df4aedf0eb887219